crates/proto/src/dnssec/dnssec_dns_handle/mod.rs

```diff
@@ -817,18 +818,36 @@ where
                 ProtoError::from(ProtoErrorKind::NoError)
             }
         }
-        Ok(_) => ProtoError::from(ProtoErrorKind::NoError),
+        Ok(response) => {
+            // If the response was an authenticated proof of nonexistence, then we have an insecure zone.
+            let empty_ds_rrset = !response
```

Comment:
Would prefer to avoid the negation inside the binding here, and using it on the condition instead. Also maybe consider inverting the condition to reduce rightward drift, and inlining the format argument.